be/src/runtime/fragment_mgr.cpp

```diff
@@ -209,9 +209,12 @@ Status FragmentExecState::execute() {
     return Status::OK();
 }
 
-Status FragmentExecState::cancel() {
+Status FragmentExecState::cancel(const PCancelPlanFragmentRequest::CancelReason& reason) {
     std::lock_guard<std::mutex> l(_status_lock);
     RETURN_IF_ERROR(_exec_status);
+    if (reason != PCancelPlanFragmentRequest::INTERNAL_ERROR) {
```

Comment:
I think it's better to use equal other than not equal here. Because when we add more types of error later, we should let it report in default.

Reply:
ok